diesel/src/macros/associations/belongs_to.rs

```diff
@@ -102,22 +101,105 @@ macro_rules! BelongsTo {
     };
 
     // Receive parsed fields of normal struct from `__diesel_parse_struct_body`
+    // Next we need to get the foreign key field in order to determine if it's optional.
     // These patterns must appear above those which start with an ident to compile
     (
         (
             struct_name = $struct_name:ident,
             parent_struct = $parent_struct:ident,
             foreign_key_name = $foreign_key_name:ident,
-            optional_foreign_key = "false",
+            $($remaining_arguments:tt)*
+        ),
+        fields = $fields:tt,
+    ) => {
+        __diesel_field_with_column_name! {
+            (
+                fields = $fields,
+                struct_name = $struct_name,
+                parent_struct = $parent_struct,
+                foreign_key_name = $foreign_key_name,
+                $($remaining_arguments)*
+            ),
+            callback = BelongsTo,
+            target = $foreign_key_name,
+            fields = $fields,
+        }
+    };
+
+    // Receive the foreign key field from __diesel_field_with_column_name!
+    (
+        (
+            fields = $fields:tt,
+            $($remaining_args:tt)*
+        ),
+        found_field_with_column_name = $ignore:ident,
+        field = {
+            field_name: $ignore2:ident,
+            column_name: $ignore3:ident,
+            field_ty: $ignore4:ty,
+            field_kind: $foreign_key_kind:ident,
```

Comment:
took me a while to see the magic to determine if this is an option is already part of `__diesel_parse_struct_body` :)